File: 0240-search-a-2d-matrix-ii/0240-search-a-2d-matrix-ii.cpp
class Solution {
public:
    bool searchMatrix(vector<vector<int>>& matrix, int target) {
        int row=matrix.size();
        int col=matrix[0].size();
        int i=0;
        int j=matrix[0].size()-1;
        while(i>=0 && j>=0 && i<row && j<col){
            if(matrix[i][j]==target){
                return true;
            }
            else if(matrix[i][j]<target){
                i++;
            }
            else{
                j--;
            }
        }
        return false;
    }
};

File: 0240-search-a-2d-matrix-ii/0240-search-a-2d-matrix-ii.java
class Solution {
    public boolean searchMatrix(int[][] matrix, int target) {
        int i = 0;
        int j = matrix[0].length-1;
        int rowSize = matrix.length;
        int colSize = matrix[0].length;
        
        while(i >= 0 && j >= 0 && i < rowSize && j < colSize){
            if(matrix[i][j] == target){
                return true;
            }
            else if(matrix[i][j] < target){
                i++;
            }
            else{
                j--;
            }
        }
        return false;
    }
}